utils.go

```diff
@@ -1,187 +1,101 @@
+	switch block.Version {
+	case consensusspec.DataVersionCapella:
+		return CapellaRequestToProtoRequest(block.Capella)
+	case consensusspec.DataVersionDeneb:
+		return DenebRequestToProtoRequest(block.Deneb)
+	default:
+		panic("unknown version")
 	}
 }
 
-func CapellaRequestToProtoRequestWithShortIDs(block *capella.SubmitBlockRequest, compressTxs []*CompressTx) *SubmitBlockRequest {
-
-	withdrawals := []*Withdrawal{}
-
-	for _, withdrawal := range block.ExecutionPayload.Withdrawals {
-		withdrawals = append(withdrawals, &Withdrawal{
-			ValidatorIndex: uint64(withdrawal.ValidatorIndex),
-			Index:          uint64(withdrawal.Index),
-			Amount:         uint64(withdrawal.Amount),
-			Address:        withdrawal.Address[:],
-		})
+// Based on the version, delegate to the correct RequestToProtoRequestWithShortIDs
+func VersionedRequestToProtoRequestWithShortIDs(block *builderSpec.VersionedSubmitBlockRequest, compressTxs []*CompressTx) *SubmitBlockRequest {
+	switch block.Version {
+	case consensusspec.DataVersionCapella:
+		return CapellaRequestToProtoRequestWithShortIDs(block.Capella, compressTxs)
+	case consensusspec.DataVersionDeneb:
+		return DenebRequestToProtoRequestWithShortIDs(block.Deneb, compressTxs)
+	default:
+		panic("unknown version")
 	}
+}
 
-	return &SubmitBlockRequest{
-		BidTrace: &BidTrace{
-			Slot:                 block.Message.Slot,
-			ParentHash:           block.Message.ParentHash[:],
-			BlockHash:            block.Message.BlockHash[:],
-			BuilderPubkey:        block.Message.BuilderPubkey[:],
-			ProposerPubkey:       block.Message.ProposerPubkey[:],
-			ProposerFeeRecipient: block.Message.ProposerFeeRecipient[:],
-			GasLimit:             block.Message.GasLimit,
-			GasUsed:              block.Message.GasUsed,
-			Value:                block.Message.Value.Hex(),
-		},
-		ExecutionPayload: &ExecutionPayload{
-			ParentHash:    block.ExecutionPayload.ParentHash[:],
-			StateRoot:     block.ExecutionPayload.StateRoot[:],
-			ReceiptsRoot:  block.ExecutionPayload.ReceiptsRoot[:],
-			LogsBloom:     block.ExecutionPayload.LogsBloom[:],
-			PrevRandao:    block.ExecutionPayload.PrevRandao[:],
-			BaseFeePerGas: block.ExecutionPayload.BaseFeePerGas[:],
-			FeeRecipient:  block.ExecutionPayload.FeeRecipient[:],
-			BlockHash:     block.ExecutionPayload.BlockHash[:],
-			ExtraData:     block.ExecutionPayload.ExtraData,
-			BlockNumber:   block.ExecutionPayload.BlockNumber,
-			GasLimit:      block.ExecutionPayload.GasLimit,
-			Timestamp:     block.ExecutionPayload.Timestamp,
-			GasUsed:       block.ExecutionPayload.GasUsed,
-			Transactions:  compressTxs,
-			Withdrawals:   withdrawals,
-		},
-		Signature: block.Signature[:],
+// Based on the version, delegate to the correct ProtoRequestToVersionedRequest
+func ProtoRequestToVersionedRequest(block *SubmitBlockRequest, version string) *builderSpec.VersionedSubmitBlockRequest {
```

Comment:
I think `version` should be a member of `SubmitBlockRequest`, that way when we use this function we don't have to check the block contents to determine which version to use

Reply:
Fixed